Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/storage/memory/GraphPath.java

```diff
@@ -0,0 +1,313 @@
+                        }
+                        String value = applyPrefixes(prefixes, string.substring(literalStart, index));
+                        result.put(currentPredicate, MemoryLiteral.createLeaf(value, null));
+                    }
+                }
+            } while (index < length);
+            return length;
+        }
+    }
+
+    /**
+     * Indicates that the object must be referenced from the subject, but the
+     * predicate of the reference is not specified.
+     */
+    public static final NodeReference ANY_PREDICATE = MemoryStorage.INSTANCE
+            .createNodeReference("http://names.kitodo.org/GraphPath/v1#anyPredicate");
+
+    /**
+     * Character in the graph path string indicating that the object must be
+     * referenced from the subject, but the predicate of the reference is not
+     * specified.
+     */
+    private static final String ANY_PREDICATE_CHAR = "*";
+
+    /**
+     * This is a graph path.
+     */
+    private static final NodeReference GRAPH_PATH = MemoryStorage.INSTANCE
+            .createNodeReference("http://names.kitodo.org/GraphPath/v1#GraphPath");
+
+    /**
+     * This is a location step of a graph path.
+     */
+    static final NodeReference LOCATION_STEP = MemoryStorage.INSTANCE
+            .createNodeReference("http://names.kitodo.org/GraphPath/v1#LocationStep");
+
+    /**
+     * Direction of reference is forwards one step.
+     */
+    static final NodeReference TO = MemoryStorage.INSTANCE
+            .createNodeReference("http://names.kitodo.org/GraphPath/v1#to");
+
+    /**
+     * Applies a graph path on a collection of nodes.
+     *
+     * <p>
+     * Currently only supports relation type “to”.
+     *
+     * @param nodes
+     *            nodes to apply the graph path on
+     * @param path
+     *            the graph path to apply
+     * @return the matching nodes
+     */
+    public static Result apply(Set<ObjectType> nodes, Node path) {
+        MemoryResult result = new MemoryResult();
+
+        Set<Node> allTo = path.get(TO).nodes();
+
+        Result nodeResult = nodes instanceof Result ? (Result) nodes : new MemoryResult();
+
+        if (allTo.isEmpty() || nodes.isEmpty()) {
+            return nodeResult;
+        }
+
+        Set<Node> nodesOnly = nodeResult.nodes();
+
+        for (Node toSegment : allTo) {
+            MemoryResult thisResult = new MemoryResult();
+            Set<String> predicates = new HashSet<>();
+            for (IdentifiableNode predicate : toSegment.get(RDF.PREDICATE).identifiableNodes()) {
+                predicates.add(predicate.getIdentifier());
+            }
+            Collection<ObjectType> objects = new LinkedList<>();
+            toSegment.get(RDF.OBJECT).forEach(objects::add);
+            for (Node node : nodesOnly) {
+                node.get(predicates, objects).forEach(thisResult::add);
+            }
+            apply(thisResult, toSegment).forEach(result::add);
+        }
+
```

Comment:
Is there are specific reason to leave here two blank lines before return and in next method none?